org.ultimatepp.TheIDE.yml

```diff
@@ -0,0 +1,115 @@
+app-id: org.ultimatepp.TheIDE
+runtime: org.gnome.Platform
+runtime-version: '45'
+sdk: org.gnome.Sdk
+sdk-extensions: ["org.freedesktop.Sdk.Extension.llvm17"]
+command: run-ide.sh
+finish-args:
+  - --socket=x11
+  - --share=network
+  - --share=ipc
+  - --filesystem=home
+  - --filesystem=host
+  - --talk-name=org.freedesktop.Flatpak
+modules:
+  - name: upp
+    buildsystem: simple
+    build-commands: 
+      - printf "#define flagFlatpak 1\\n" > uppsrc/uppconfig.h
+      - git apply flatpak.patch
+      - make -f Makefile -j $FLATPAK_BUILDER_N_JOBS
+      - mv ide theide
+      - install -D theide /app/bin/theide
+      - mkdir /app/src
+      - mkdir /app/lib
+      - printf "\\n" > uppsrc/uppconfig.h
+      - rm -f /app/src/ver.txt && touch /app/src/ver.txt
+      - printf "2023.2" > /app/src/ver.txt
+      - cp -r examples /app/src
+      - cp -r reference /app/src
+      - cp -r tutorial /app/src
+      - cp -r uppsrc /app/src
+      - cp .clang-format /app/src/uppsrc/.clang-format
+      - cp run-ide-install-host-deps.sh /app/src/run-ide-install-host-deps.sh
+      - cp -r /usr/lib/sdk/llvm17/lib /app
```

Comment:
why do you copy this?

Reply:
We are using libclang for auto complete. We load libclang.so during runtime. For more info please look at this line https://github.com/ultimatepp/ultimatepp/blob/ef6d9db3de5db17b0ed3d5c55bf9d6812fe7f1eb/uppsrc/ide/clang/libclang.cpp#L30.

Without this line the auto complete feature (we call it assist++) doesn't work in flatpak sandbox environment.